plugins/browser-plugin-link-click-tracking/src/index.ts

```diff
@@ -83,178 +99,200 @@ export interface LinkClickTrackingConfiguration {
+ *
+ * @deprecated v4.0 moved to event delegation and this is no longer required
+ * @param trackers The tracker identifiers which the have their link click state refreshed
+ */
+export function refreshLinkClickTracking(_trackers: Array<string> = []) {
+  _logger?.warn('refreshLinkClickTracking is deprecated in v4 and has no effect');
+}
+
+/**
+ * Manually log a click.
  *
- * @param event - The event information
- * @param trackers - The tracker identifiers which the event will be sent to
+ * @param event The event information
+ * @param trackers The tracker identifiers which the event will be sent to
  */
 export function trackLinkClick(
-  event: LinkClickEvent & CommonEventProperties,
+  event: (LinkClickEvent | { element: TrackableElement }) & CommonEventProperties,
   trackers: Array<string> = Object.keys(_trackers)
 ) {
   dispatchToTrackersInCollection(trackers, _trackers, (t) => {
-    t.core.track(buildLinkClick(event), event.context, event.timestamp);
+    let payload: PayloadBuilder | undefined;
+    if ('element' in event) {
+      const includeContent = _configuration[t.id]?.linkTrackingContent ?? false;
+      payload = processClick(event.element, includeContent);
+    } else {
+      payload = buildLinkClick(event);
+    }
+    if (payload) t.core.track(payload, event.context, event.timestamp);
   });
 }
 
-/*
- * Process clicks
+/**
+ * Process a clicked element into a link_click event payload.
+ *
+ * @param sourceElement The trackable element to be used to build the payload
+ * @param includeContent Whether to include the element's contents in the payload
+ * @returns A link_click SDE payload for the given element, or nothing if the element shouldn't be tracked
  */
-function processClick(tracker: BrowserTracker, sourceElement: Element, context?: DynamicContext | null) {
-  let parentElement, tag, elementId, elementClasses, elementTarget, elementContent;
+function processClick(sourceElement: TrackableElement, includeContent: boolean = false) {
+  let elementId, elementClasses, elementTarget, elementContent;
+
+  const anchorElement = sourceElement;
+  // browsers, such as Safari, don't downcase hostname and href
+  const originalSourceHostName = anchorElement.hostname || getHostName(anchorElement.href);
+  let sourceHref = anchorElement.href.replace(originalSourceHostName, (s) => s.toLowerCase());
 
-  while (
-    (parentElement = sourceElement.parentElement) !== null &&
-    parentElement != null &&
-    (tag = sourceElement.tagName.toUpperCase()) !== 'A' &&
-    tag !== 'AREA'
-  ) {
-    sourceElement = parentElement;
+  // Ignore script pseudo-protocol links
+  if (!ELEMENT_PROTOCOL_FILTER.test(sourceHref)) {
+    elementId = anchorElement.id;
+    elementClasses = getCssClasses(anchorElement);
+    elementTarget = anchorElement.target;
+    elementContent = includeContent ? anchorElement.innerHTML : undefined;
+
+    // decodeUrl %xx
+    // NOTE(jethron): Why do we do this?
```

Comment:
I think it would be a good idea to drop the `unescape` or replace it with `decodeURI`. Based [on the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape), they don't recommend using it.

I assume what we want to do here is just to make the URI look nicer so that a URL like:
`https://snowplow.io/?x=a%20b%20c`
will turn into:
`https://snowplow.io/?x=a b c`

I am not convinced it's a good idea, because we don't seem to do it for the URL property in page views, which creates an inconsistency in how the URL is represented.

I think that as part of the breaking changes of the v4, we can also drop this. What do you think?

Reply:
Yeah I agree; it's a lossy transform and would be hard to parse after the fact for any non-human-legibility use case. I'll remove this.